zabbix_auto_config/models.py

```diff
@@ -390,6 +400,13 @@ class SourceCollectorSettings(ConfigBaseModel, extra="allow"):
         ge=0,
     )
 
+    model_config = ConfigDict(
+        # Validators cause infinte recursion if assignment validation is enabled.
+        # TODO: fix
+        validate_assignment=False,
+        extra="allow",
+    )
```

Comment:
Having to add `validate_assignment=False` in order to avoid infinite recursion, due to validator mutating a value, which then calls itself and mutates itself again, is a sign of some seriously bad code.